### PR TITLE
handle sort order on library menu resources

### PR DESCRIPTION
**Pull request by MrGrinst**
Handles sort order when viewing by resource on the Library. Also handled a couple small things I noticed with debouncing and the situation where search returns nothing.

### CODE REVIEW

File: src/routes/view-content/[guideId]/[bibleSection]/library-resource-menu/SearchByResource.svelte [hunk 1]
```diff
@@ -28,7 +30,11 @@
     let currentParentResource: ApiParentResource | null;
     let searchQuery = '';
 
-    $: searchByParentResource(currentParentResource, 0, searchQuery, false);
+    $: debouncedSearchByParentResource(searchQuery);
+
+    const debouncedSearchByParentResource = debounce(async (query: string) => {
+        searchByParentResource(currentParentResource, 0, query, false);
+    }, 750);
```
Comment (MrGrinst) [inline src/routes/view-content/[guideId]/[bibleSection]/library-resource-menu/SearchByResource.svelte:37]: Also added a debounce here so we don't query every keystroke

File: src/routes/view-content/[guideId]/[bibleSection]/library-resource-menu/SearchByResource.svelte [hunk 2]
```diff
@@ -144,6 +150,8 @@
                         class="mx-4 mb-4 flex h-12 items-center justify-center rounded-lg border border-[#EAECF0] p-2 text-sm"
                         id={`resource-button-${index + 1}`}>{resourceSearchResource.displayName}</button
                     >
+                {:else}
+                    <NoResourcesFound {searchQuery} />
```
Comment (MrGrinst) [inline src/routes/view-content/[guideId]/[bibleSection]/library-resource-menu/SearchByResource.svelte:154]: And added handling for no resources found

File: src/lib/utils/data-handlers/resources/resource.ts [hunk 3]
```diff
@@ -237,10 +237,17 @@ export async function filterToAvailableAssociatedResourceContent(
...
-export function sortByDisplayName<T extends { displayName: string | undefined }>(resources: T[]): T[] {
-    return resources.sort((a, b) =>
-        (a.displayName ?? '').localeCompare(b.displayName ?? '', undefined, { numeric: true })
-    );
+export function sortByDisplayName<T extends { displayName: string | undefined; sortOrder?: number }>(
+    resources: T[]
+): T[] {
+    return resources.sort((a, b) => {
+        if (a.sortOrder !== undefined && b.sortOrder !== undefined) {
+            if (a.sortOrder !== b.sortOrder) {
```
Comment (jwinston-bn) [inline src/lib/utils/data-handlers/resources/resource.ts:245]: Does this need to be an additional `if` as opposed to just `&&` in the above?

**Review by carlosmorales125 — APPROVED** [review]
Thanks for doing this.

File: src/lib/utils/data-handlers/resources/resource.ts (hunk 3 above)
Comment (MrGrinst) [reply src/lib/utils/data-handlers/resources/resource.ts:245]: Probably not but I had set to auto-merge, I'll address in my next PR

**Pull request merged**